### PR TITLE
dnd-character: test hitpoints value exactly

### LINKED ISSUE #591: D&D: Add a test to ensure that hitpoints is calculated from constitution
I saw a solution where `hitpoints` was a random `ability` value instead of `10 + modifier(constitution)`. We should add a test for this. The existing test is not that precise.

### PR DESCRIPTION
**Pull request by glennj**
closes #591

### CODE REVIEW

File: exercises/practice/dnd-character/.meta/example.sh [hunk 1]
```diff
@@ -32,7 +32,7 @@ generate() {
     local a c hp
     for c in "${characteristics[@]}"; do
         a=$(ability)
-        printf "%s %d\n" "$c" "$(ability)"
+        printf "%s %d\n" "$c" "$a"
```
Comment (glennj) [inline exercises/practice/dnd-character/.meta/example.sh:35]: This was a bug.

File: exercises/practice/dnd-character/dnd_character.bats [hunk 2]
```diff
@@ -131,34 +131,39 @@ load bats-extra
...
-    # and no other output
-    assert_equal "$(echo "$output" | wc -l)" 7
+    # these don't have to appear in any particular order
+    assert_line --regexp '^strength [[:digit:]]{1,2}$'
+    assert_line --regexp '^dexterity [[:digit:]]{1,2}$'
+    assert_line --regexp '^constitution [[:digit:]]{1,2}$'
+    assert_line --regexp '^intelligence [[:digit:]]{1,2}$'
+    assert_line --regexp '^wisdom [[:digit:]]{1,2}$'
+    assert_line --regexp '^charisma [[:digit:]]{1,2}$'
+    assert_line --regexp '^hitpoints [[:digit:]]{1,2}$'
```
Comment (glennj) [inline exercises/practice/dnd-character/dnd_character.bats:141]: This is just making the regex a bit more specific.

File: exercises/practice/dnd-character/dnd_character.bats [hunk 3]
```diff
@@ -131,34 +131,39 @@ load bats-extra
...
+    # no other output: `run` populates the `lines` array
+    assert_equal ${#lines[@]} 7
```
Comment (glennj) [inline exercises/practice/dnd-character/dnd_character.bats:143]: Simplify

File: exercises/practice/dnd-character/dnd_character.bats [hunk 4]
```diff
@@ -131,34 +131,39 @@ load bats-extra
...
 }
 
 
 # Usage: between $val $low $high
-# Value is between low (inclusive) and high (exclusive).
+# Value is between low (inclusive) and high (inclusive).
 between() { 
-    (( $2 <= $1 && $1 < $3 ))
+    (( $2 <= $1 && $1 <= $3 ))
```
Comment (glennj) [inline exercises/practice/dnd-character/dnd_character.bats:150]: This function isn't really needed, but I like it!

File: exercises/practice/dnd-character/dnd_character.bats [hunk 5]
```diff
@@ -131,34 +131,39 @@ load bats-extra
...
             if [[ $c == "hitpoints" ]]; then
-                assert between "$v" 6 15
+                hits=$v
             else
-                assert between "$v" 3 19
+                assert between "$v" 3 18
+                [[ $c == "constitution" ]] && const=$v
             fi
-        done < <( run bash dnd_character.sh generate )
+        done < <(bash dnd_character.sh generate)
```
Comment (glennj) [inline exercises/practice/dnd-character/dnd_character.bats:164]: This removes `run` -- I don't know how it worked before.

**Review by IsaacG — APPROVED** [review]

**Review by kotp — APPROVED** [review]

**Pull request merged**